edb/server/compiler_pool/pool.py

```diff
@@ -736,6 +703,201 @@ async def describe_database_restore(
             self._release_worker(worker)
 
 
+@srvargs.CompilerPoolScalingMode.Fixed.set_class
+class FixedPool(BasePool):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self._template_transport = None
+        self._template_proc_scheduled = False
+        self._template_proc_version = 0
+
+    def _worker_attached(self):
+        if len(self._workers) > self._pool_size:
+            self._server.kill_outdated_worker(self._template_proc_version)
+
+    def worker_connected(self, pid, version):
+        if version < self._template_proc_version:
+            logger.debug(
+                "Outdated worker with PID %s connected; discard now.", pid
+            )
+            self._server.get_by_pid(pid).abort()
+            metrics.compiler_process_spawns.inc()
+        else:
+            super().worker_connected(pid, version)
+
+    def process_exited(self):
+        # Template process exited
+        self._template_transport = None
+        if self._running:
+            logger.error("Template compiler process exited; recreating now.")
+            self._schedule_template_proc(0)
+
+    def get_template_pid(self):
+        if self._template_transport is None:
+            return None
+        else:
+            return self._template_transport.get_pid()
+
+    async def _start(self):
+        await self._create_template_proc(retry=False)
+
+    async def _create_template_proc(self, retry=True):
+        self._template_proc_scheduled = False
+        if not self._running:
+            return
+        self._template_proc_version += 1
+        try:
+            self._template_transport = await self._create_worker_process(
```

Comment:
It's a bit hard to follow the changes -- are we still using fork for the "fixed" pool?

Reply:
This is in part because of using inheritance. Can you add comments here and there explaining the key points: where we spawn, where we fork, and how we do that?

Reply:
Yeah sure, np